src/v/cloud_storage/topic_manifest.cc

```diff
@@ -483,7 +523,11 @@ void topic_manifest::serialize(std::ostream& out) const {
         }
     }
 
-    if (_topic_config->properties.mpx_virtual_cluster_id) {
+    // do not serialize fields that are not deserializable by previous versions
+    // of redpanda
```

Comment:
Just curious, is there any harm in serializing this field? During a partial upgrade is it possible for this to be set, which doesn't immediately seem problematic, so why bother with this condition? Though I may be missing something

Reply:
if you remember a previous discussion, topic_manifest_handler will treat unknown fields as an error (part of the reason this pr has been so not linear). so (it's an edge case) in a mixed version cluster, if we serialize this field, then we risk side effects in the older nodes. 

It probably can be unblocked with a backport removing the hard requirement, but it's not a complete solution.

I debated if to remove mpx_virtual_cluster_id, but it feels too hazardous right now.

Reply:
I see. This field hasn't been released yet AFAICT https://github.com/redpanda-data/redpanda/pull/16714

Why bother ever serializing as JSON then?

Reply:
don't want to make @mmaslankaprv angry 👀

(@mmaslankaprv since we are moving to a serde topic manifest, it should be ok to not serialize mpx_virtual_cluster_id to json here, right?)

Reply:
I think the case where there is a material difference is if the topic is assigned a virtual cluster id, and then we go through a partial upgrade. While that partial upgrade is happening, we may not yet bet serializing with serde, and may upload JSON with this virtual cluster id.

From #16714 it isn't very clear whether it's expected that we may have mixed-version clusters with this property in the first place (mechanically, it seems possible, but since this is part of a broader more nuanced deployment style, perhaps there is something that prevents it)

Reply:
keeping this like this for now